src/server/pfs/drive/driver.go

```diff
@@ -592,10 +713,35 @@ func filterBlockRefs(filterShard *pfs.Shard, blockRefs []*pfs.BlockRef) []*pfs.B
 	return result
 }
 
-func (d *driver) inspectFile(file *pfs.File, filterShard *pfs.Shard, shard uint64, from *pfs.Commit) (*pfs.FileInfo, []*pfs.BlockRef, error) {
+func (d *driver) getFileType(file *pfs.File, shard uint64) (pfs.FileType, error) {
+	commit, err := d.canonicalCommit(file.Commit)
+	if err != nil {
+		return pfsclient.FileType_FILE_TYPE_NONE, err
+	}
+	for commit != nil {
+		diffInfo, ok := d.diffs.get(pfsclient.NewDiff(commit.Repo.Name, commit.ID, shard))
+		if !ok {
+			return pfs.FileType_FILE_TYPE_NONE, fmt.Errorf("diff %s/%s/%d not found", commit.Repo.Name, commit.ID, shard)
+		}
+		if _append, ok := diffInfo.Appends[path.Clean(file.Path)]; ok {
+			if _append.Delete {
+				break
+			} else if len(_append.BlockRefs) > 0 {
+				return pfs.FileType_FILE_TYPE_REGULAR, nil
+			} else {
+				return pfs.FileType_FILE_TYPE_DIR, nil
+			}
+		}
+		commit = diffInfo.ParentCommit
+	}
+	return pfs.FileType_FILE_TYPE_NONE, nil
+}
+
+func (d *driver) inspectFile(file *pfs.File, filterShard *pfs.Shard, shard uint64, from *pfs.Commit, unsafe bool) (*pfs.FileInfo, []*pfs.BlockRef, error) {
```

Comment:
Would you mind adding a comment here about what each parameter does?

It should have already had one.